composites/AlgoliaSearch/AlgoliaSearcher.js

```diff
@@ -141,182 +168,147 @@ class AlgoliaSearcher extends React.Component {
-			usedQueries: usedQueries,
-		} );
+		return usedQueries;
 	}
 
 	/**
-	 * Hides the details page and return to the results page.
+	 * Sets all values required to display the detailed view of a search result.
 	 *
-	 * @returns {void}
-	 */
-	hideDetail() {
-		this.setState( { showDetail: false } );
-	}
-
-	/**
-	 * Renders a no results found text.
+	 * @param {number} currentDetailViewIndex The index of the article you want to show in the state.results array.
 	 *
-	 * @returns {ReactElement} The no results found text.
+	 * @returns {void}
 	 */
-	renderNoResultsFound() {
-		let searchResultContent = <p>{ this.props.noResultsText }</p>;
-		a11ySpeak( this.props.noResultsText );
+	setCurrentDetailedViewIndex( currentDetailViewIndex ) {
+		let usedQueries = this.addResultToUsedQueries( currentDetailViewIndex );
 
-		return searchResultContent;
+		this.setState( { currentDetailViewIndex, usedQueries } );
 	}
 
 	/**
-	 * Maps the results to SearchResult components.
-	 *
-	 * @param {object} results The results returned by Algolia.
+	 * Resets the values associated with the displaying of the detailed search result view.
 	 *
-	 * @returns {Array} Array containing the mapped search results.
+	 * @returns {void}
 	 */
-	resultsToSearchItem( results ) {
-		return results.map( ( result, index ) => {
-			return <SearchResult
-				key={ result.objectID }
-				post={ result }
-				showDetail={ this.showDetail.bind( this, index ) }
-			/>;
-		} );
+	unsetCurrentDetailedView() {
+		this.setState( { currentDetailViewIndex: -1 } );
 	}
 
 	/**
-	 * Renders the search results list.
+	 * Log any occuring error and render a search error warning.
 	 *
-	 * @returns {ReactElement|null} A div with either the search results, or a div with a message that no results were found.
+	 * @returns {ReactElement} A p element with a warning that the search was not completed.
 	 */
-	renderSearchResults() {
-		let searchResultContent = null;
-		let resultsCount = this.state.results.length;
-
-		// We'll check to see whether no results are returned.
-		if ( resultsCount <= 0 && this.state.searchString !== "" ) {
-			return this.renderNoResultsFound();
-		}
+	renderError() {
+		const errorMessage = this.props.intl.formatMessage( messages.errorMessage );
 
-		if ( resultsCount === 0 ) {
-			return searchResultContent;
-		}
-
-		let results = this.resultsToSearchItem( this.state.results );
-
-		searchResultContent = <ul role="list" className="wpseo-kb-search-results">{ results }</ul>;
-		a11ySpeak( this.props.foundResultsText.replace( "%d", resultsCount ) );
+		a11ySpeak( errorMessage );
 
-		return searchResultContent;
+		return (
+			<p>{ errorMessage }</p>
+		);
 	}
 
 	/**
-	 * Renders the navigation buttons and the article content.
+	 * Creates the Search Bar component.
 	 *
-	 * @returns {ReactElement} A div with navigation buttons and the article content.
+	 * @returns {ReactElement|string} Searchbar component. Returns empty string if we're in the detail view.
```

Comment:
technically, "Returns empty string if ..." is not correct, because it returns `null`.